src/SW_netCDF.c

```diff
@@ -0,0 +1,591 @@
+                  domainID, LogInfo);
+
+    // "site" variable
+    create_variable("site", ONEDIM, dims, NC_INT, &sID, LogInfo);
+    write_att_str("units", "1", sID, LogInfo);
+    write_att_str("long_name", "SOILWAT2 simulation sites", sID, LogInfo);
+
+    // "x" variable
+    create_var_x(&xID, dims, LogInfo);
+    nc_put_att_uint(OPEN_NC_ID, xID, "_ChunkSize", NC_UINT, numChunkVals,
+                    ChunkVals);
+    write_att_val("_FillValue", fillVal, numChunkVals, xID, LogInfo);
+
+    // "y" variable
+    create_var_y(&yID, dims, LogInfo);
+    nc_put_att_uint(OPEN_NC_ID, yID, "_ChunkSize", NC_UINT, numChunkVals,
+                    ChunkVals);
+    nc_put_att_double(OPEN_NC_ID, yID, "_FillValue", NC_DOUBLE, numChunkVals,
+                    fillVal);
+
+    nc_enddef(OPEN_NC_ID); // End of defining variables, dimensions, etc.
+
+    fill_s_vars(nDimS, sID, domainID, xID, yID);
+}
+
+/**
+ * @brief Create an "x" variable with common attributes between the domains
+ *  "xy" and "site"
+ *
+ * @param[in] xID ID of the x variable within the domain netCDF
+ * @param[in] xDim Dimensions of the x variable
+ * @param[in] LogInfo Holds information dealing with logfile output
+*/
+void create_var_x(int* xID, int xDim[], LOG_INFO* LogInfo) {
+
+    create_variable("x", ONEDIM, xDim, NC_DOUBLE, xID, LogInfo);
+
+    write_att_str("units", "m", *xID, LogInfo);
+    write_att_str("long_name", "projection_x_coordinate", *xID, LogInfo);
+    write_att_str("standard_name", "x coordinate of projection", *xID, LogInfo);
+}
+
+/**
+ * @brief Create a "y" variable with common attributes between the domains
+ *  "xy" and "site"
+ *
+ * @param[in] yID ID of the y variable within the domain netCDF
+ * @param[in] yDim Dimensions of the y variable
+ * @param[in] LogInfo Holds information dealing with logfile output
+*/
+void create_var_y(int *yID, int yDim[], LOG_INFO* LogInfo) {
+
+    create_variable("y", ONEDIM, yDim, NC_DOUBLE, yID, LogInfo);
+
+    write_att_str("units", "m", *yID, LogInfo);
+    write_att_str("long_name", "projection_y_coordinate", *yID, LogInfo);
+    write_att_str("standard_name", "y coordinate of projection", *yID, LogInfo);
+}
+
+/**
+ * @brief Create a "domain" variable with common attributes between the domains
+ *  "xy" and "site"
+ *
+ * @param[in] domID ID of the y variable within the domain netCDF
+ * @param[in] domDim Dimensions of the domain variable
+ * @param[in] numDims The number of dimensions "domain" will be
+ * @param[in] chunkSize How many values that will be written under the
+ *  "_ChunkSizes" attribute
+ * @param[in] chunkVals Values under the attribute "_ChunkSizes"
+ * @param[in] LogInfo Holds information dealing with logfile output
+*/
+void create_var_domain(int* domID, int domDim[], int numDims,
+        size_t chunkSize, unsigned int chunkVals[], LOG_INFO* LogInfo) {
+
+    // Longer attribute names
+    int numFillSize = 1;
+    float fillVal[] = {-3.4E38};
```

Comment:
Let's use the "default" fill value of `NC_FILL_FLOAT` if variable type is `NC_FLOAT`. See https://docs.unidata.ucar.edu/netcdf-c/current/netcdf_8h.html#ad9a1c544d512d09c610e181f0709885a